lib/packwerk/cli.rb

```diff
@@ -1,15 +1,29 @@
 # typed: strict
 # frozen_string_literal: true
 
-require "optparse"
-
 module Packwerk
   # A command-line interface to Packwerk.
   class Cli
     extend T::Sig
     extend ActiveSupport::Autoload
 
-    autoload :Result
+    autoload :CommandRegistry
+    autoload :BaseCommand
+    autoload :CheckCommand
+    autoload :HelpCommand
+    autoload :InitCommand
+    autoload :UpdateTodoCommand
+    autoload :UsesParseRun
+    autoload :ValidateCommand
+    autoload :VersionCommand
+
+    class << self
+      extend T::Sig
+      sig { params(name: String, help: String, aliases: T::Array[String]).void }
+      def register_command(name, help:, aliases: [])
+        CommandRegistry.register(name, help: help, aliases: aliases)
```

Comment:
This just delegates to `CommandRegistry`. It allows me to keep that class private, while it exposes public methods to be used within Packwerk.